### PR TITLE
[AC-xx: CI-CD] Kelvin - add Jenkinsfile, Dockerfile and eks yaml.

### PR DESCRIPTION
**Pull request by NubKel**
### Provide descriptions of this pull request.
Add Dockerfile, eks yaml file for future deployment purposes, Jenkinsfile with a pipeline including code checks, backup image to ECR and deploy frontend service in ECS.

### Provide screenshots of no warning, error in console.
image
The pipeline works as expected.

image
The docker image uploads to ECR.

image
The service deploys to ECS.

- [x] No merge conflict with the `develop` branch.
- [x] Check PR title is correct, (i.e `[AC-xx: Feature] Create pull request template`).
- [x] Check your app will run without crashes.
- [x] Check if there are hard coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] No unnecessary spaces between lines of code.
- [ ] Check you are using `camelCase && PascalCase` correctly.
- [x] Check you named your code properly.
- [ ] Remove unused comments.

### CODE REVIEW

File: CI-CD/Jenkinsfile.Kelvin [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            }
+
+            steps {
+                withAWS(credentials: 'AWS_CREDENTIAL', region: 'ap-southeast-2') {
+                    sh 'aws ecs update-service --cluster ${ECS_CLUSTER} --service ${ECS_SERVICE} --force-new-deployment'
+                } 
+            }
+        }
+
+        // stage ('Deploying to EKS...') {
```
Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Kelvin]: 注释去掉

File: CI-CD/Jenkinsfile.Kelvin [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    }
+                }
+            }
+        }
+
+        stage ('Deploying to ECS...') {
+            when {
+                anyOf {
+                    branch 'develop'
+                    branch 'ci-cd/*'
```
Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Kelvin]: 改成精确触发你的branch

File: CI-CD/Jenkinsfile.Kelvin [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        //     steps {
+        //         kubernetesDeploy(
+        //             configs: 'CI-CD/kel-eks.yaml',
+        //             kubeconfigId: 'K8S',
+        //             enableConfigSubstitution: true
+        //         )
+        //     }
+        // }
+            
+    }
```
Comment (tardigrade425) [inline CI-CD/Jenkinsfile.Kelvin:93]: 加上 Post

File: Dockerfile.Kelvin [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# Install dependencies only when needed
+FROM node:16-alpine AS deps
+# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
+COPY package.json yarn.lock ./
+RUN yarn install --frozen-lockfile
+
+# If using npm with a `package-lock.json` comment out above and use below instead
```
Comment (tardigrade425) [inline Dockerfile.Kelvin]: 没用的注释去了

**Review by tardigrade425 — CHANGES_REQUESTED** [review]

File: CI-CD/Jenkinsfile.Kelvin (hunk 2 above)
Comment (NubKel) [reply CI-CD/Jenkinsfile.Kelvin]: 這個可以保留develop吧? CICD的branch先刪除? 因為這個branch一旦merge之後specify當前的branch也沒意義了

File: CI-CD/Jenkinsfile.Kelvin (hunk 1 above)
Comment (NubKel) [reply CI-CD/Jenkinsfile.Kelvin]: Done

File: CI-CD/Jenkinsfile.Kelvin (hunk 3 above)
Comment (NubKel) [reply CI-CD/Jenkinsfile.Kelvin:93]: Done

File: Dockerfile.Kelvin (hunk 4 above)
Comment (NubKel) [reply Dockerfile.Kelvin]: Done

File: CI-CD/Jenkinsfile.Kelvin (hunk 2 above)
Comment (tardigrade425) [reply CI-CD/Jenkinsfile.Kelvin]: 如果有其他ci-cd开头的branch，会把其他人的也build

**Review by tardigrade425 — APPROVED** [review]

**Pull request merged**